cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/EventSubscriber.java

```diff
@@ -0,0 +1,95 @@
+package io.cdap.cdap.internal.events;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import io.cdap.cdap.api.messaging.Message;
+import io.cdap.cdap.api.messaging.MessagingContext;
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.service.RetryStrategies;
+import io.cdap.cdap.common.utils.ImmutablePair;
+import io.cdap.cdap.internal.app.store.AppMetadataStore;
+import io.cdap.cdap.messaging.MessagingService;
+import io.cdap.cdap.messaging.context.MultiThreadMessagingContext;
+import io.cdap.cdap.messaging.subscriber.AbstractMessagingSubscriberService;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import org.apache.tephra.TxConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.Iterator;
+
+public class EventSubscriber extends AbstractMessagingSubscriberService<Event<?>> {
+    private static final Logger LOG = LoggerFactory.getLogger(EventSubscriber.class);
+    //TODO Determinate events topic name for TMS
```

Comment:
https://github.com/cdapio/cdap/blob/40cf403a7691742d93d836c35036d7d12284a504/cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java#L317 - This topic should give the programs status event.